src/errors.js

```diff
@@ -1,12 +1,22 @@
-class EncryptionError extends Error {}
-class EncryptionJsonError extends EncryptionError {}
-class DecryptionError extends Error {}
-class DecryptionIntegrityError extends DecryptionError {}
-class DecryptionSignatureError extends DecryptionError {}
-class DecryptionTimeTravelError extends DecryptionError {}
-class DecryptionExpiredError extends DecryptionError {}
-class DecryptionGunzipError extends DecryptionError {}
-class DecryptionJsonError extends DecryptionError {}
+class NamedError extends Error {
+  constructor(message, fileName, lineNumber) {
+    super(message, fileName, lineNumber);
+    this.message = this.message || this.constructor.name;
+  }
+
+  toJSON() {
+    return { message: this.message };
```

Comment:
String would be valid json. Is there a reason you're wrapping in `message` and not just returning it as a string?

Trying to stay true to the convention of `String(new Error('hi'));` I would expect `` `Error: ${this.message}` ``.

Reply:
Before my changes error used to be serialized to JSON as an object,
now it remains an object, but not empty, so I tried to change less.